workflows/dnaseq/dnaseq-core.wdl

```diff
@@ -0,0 +1,109 @@
+        prefix: "Prefix for the BAM file. The extension `.bam` will be added."
+        aligner: {
+            description: "BWA aligner to use",
+            choices: ["mem", "aln"]
+        }
+        validate_input: "Ensure input BAM is well-formed before beginning harmonization?"
+        use_all_cores: "Use all cores? Recommended for cloud environments."
+    }
+    input {
+        Array[File] read_one_fastqs_gz
+        Array[File] read_two_fastqs_gz
+        File bwa_db
+        Int reads_per_file = 10000000
+        Array[ReadGroup] read_groups
+        String prefix
+        String aligner = "mem"
+        Boolean use_all_cores = false
+    }
+
+    scatter (rg in read_groups) {
+        call read_group.ReadGroup_to_string { input: read_group = rg }
+    }
+
+    Array[String] read_groups_bwa = prefix("@RG ", ReadGroup_to_string.stringified_read_group)
+
+    scatter (tuple in zip(
+        zip(read_one_fastqs_gz, read_two_fastqs_gz),
+        read_groups_bwa
+    )) {
+        call util.split_fastq as read_ones { input:
+            fastq = tuple.left.left,
+            reads_per_file = reads_per_file
+        }
+
+        call util.split_fastq as read_twos { input:
+            fastq = select_first([tuple.left.right, "undefined"]),
+            reads_per_file = reads_per_file
+        }
+
+        scatter (t in zip(read_ones.fastqs, read_twos.fastqs)) {
+            if (aligner == "mem") {
+                call bwa.bwa_mem { input:
+                    read_one_fastq_gz = t.left,
+                    read_two_fastq_gz = select_first([t.right, "undefined"]),
+                    bwa_db_tar_gz = bwa_db,
+                    # find tab literals, replace with '\\t' (which must be written as '\\\\t')
+                    # '\\t' is subbed into command blocks as '\t'
+                    read_group = sub(tuple.right, " ", "\\\\t"),
+                    use_all_cores,
+                }
+            }
+            if (aligner == "aln") {
+                call bwa.bwa_aln_pe { input:
+                    read_one_fastq_gz = t.left,
+                    read_two_fastq_gz = select_first([t.right, "undefined"]),
+                    bwa_db_tar_gz = bwa_db,
+                    # find tab literals, replace with '\\t' (which must be written as '\\\\t')
+                    # '\\t' is subbed into command blocks as '\t'
+                    read_group = sub(tuple.right, " ", "\\\\t"),
+                    use_all_cores,
+                }
+            }
+            call picard.sort as sort { input:
+                 bam = select_first([bwa_mem.bam, bwa_aln_pe.bam])
+            }
+        }
+    }
+    call samtools_merge_wf.samtools_merge as rg_merge { input:
+        bams = flatten(sort.sorted_bam),
```

Comment:
```suggestion
        bams = sort.sorted_bam,
```
I'm the one who initially suggested `flatten()` but I don't think it's needed here.

Reply:
There is a `scatter` within a `scatter` above, so `sort.sorted_bam` is `Array[Array[File]]`.